both/quizModel.js

```diff
@@ -0,0 +1,197 @@
+Quiz = function(){
+    var quiz = this;
+    this.addNewQuestion = function(val){
+        //check if the question already exists
+        if (quiz.questions == undefined){
+            quiz.questions = [];
+        }
+        for (var q in quiz.questions){
+
+        };
+
+        quiz.questions.push(val);
+    };
+
+};
+
+Quiz.convertToQuizObject = function(object){
+
+    if (object == undefined) return;
+    var quiz = new Quiz();
+
+    quiz._id = object._id;
+    quiz.title = object.title;
+    quiz.lessonID = object.lessonID;
+    quiz.questions = object.questions;
+    quiz.userAttempts = object.userAttempts;
+
+    return quiz;
+};
+
+Quiz.generateQuestion = function(questionType, quizId ){
+    var question = new Object();
+    question.quizId = quizId;
+    question.id = Random.id(); //assign an id to the question - need this for checking and validating answers
+    question.questionType = questionType;
+
+    //if the question type is true-or-false, populate the answer options
+    if (question.questionType == QuizOptions.TRUE_OR_FALSE){
```

Comment:
We can use SimpleSchema to dynamically define what field to display, based on the `question.type` field.